ThePensionsRegulator.Frontend/wwwroot/tpr/headermenu.js

```diff
@@ -134,6 +134,12 @@ function displayDesktopOverlayOnHover() {
     });
 }
 
+function removeOverlayOnClick() {
+
+    const overlay = document.querySelector(".tpr-header-menu__nav-overlay");
```

Comment:
I'm not sure if this is what you meant by "Able to close the overlay with click in desktop view" but if complete the following steps the overlay remains:

1. load the example page https://localhost:44350/the-pensions-regulator-tpr-header/ and scroll to the bottom
2. position my mouse below the menu
3. use the keyboard to open a submenu, thereby displaying the overlay
4. click on the overlay

I think this is a pretty unlikely real-world scenario as a mouse user is unlikely to be navigating the menu with the keyboard, but I wasn't sure what other scenario you were trying to fix.